### PR TITLE
[ADAM-1006] Split überjar out to adam-assembly submodule.

### LINKED ISSUE #1006: Separate fat jar build from adam-cli to new maven module
Downstream projects may wish to depend on ```adam-cli``` without depending on the fat jar built by the maven shade plugin.

This would prevent warnings such as
```bash
[WARNING] adam-cli_2.10-0.19.1-SNAPSHOT.jar, commons-codec-1.3.jar define 24 overlapping classes: 
[WARNING]   - org.apache.commons.codec.StringEncoderComparator
[WARNING]   - org.apache.commons.codec.DecoderException
[WARNING]   - org.apache.commons.codec.EncoderException
[WARNING]   - org.apache.commons.codec.language.SoundexUtils
[WARNING]   - org.apache.commons.codec.net.BCodec
[WARNING]   - org.apache.commons.codec.StringDecoder
[WARNING]   - org.apache.commons.codec.language.DoubleMetaphone$DoubleMetaphoneResult
[WARNING]   - org.apache.commons.codec.language.Soundex
[WARNING]   - org.apache.commons.codec.net.RFC1522Codec
[WARNING]   - org.apache.commons.codec.BinaryDecoder
[WARNING]   - 14 more...
[WARNING] jackson-core-asl-1.9.13.jar, adam-cli_2.10-0.19.1-SNAPSHOT.jar define 121 overlapping classes: 
[WARNING]   - org.codehaus.jackson.annotate.JsonManagedReference
...
```

See also recent major changes to Spark's build that we may need to consider
https://github.com/apache/spark/commit/24d7d2e453ab5eef6099a32fb9e8ed60f6ada93a

### PR DESCRIPTION
**Pull request by fnothaft**
Resolves #1006.

### CODE REVIEW

File: adam-assembly/pom.xml [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    </plugins>
+  </build>
+  <dependencies>
+    <dependency>
+      <groupId>org.bdgenomics.adam</groupId>
+      <artifactId>adam-cli_2.10</artifactId>
+    </dependency>
+  </dependencies>
+  <profiles>
+    <profile>
```
Comment (heuermh) [inline adam-assembly/pom.xml]: this can be removed

**Comment by fnothaft** [comment]
Jenkins, retest this please.

**Comment by fnothaft** [comment]
Jenkins, retest this please.

**Comment by fnothaft** [comment]
Jenkins, retest this please.

**Comment by fnothaft** [comment]
Jenkins, retest this please.

**Comment by fnothaft** [comment]
@heuermh I believe this is good to go.

File: bin/adam-submit [hunk 2]
```diff
@@ -57,17 +57,17 @@ ADAM_CLI_JAR=
...
-num_jars="$(ls -1 "$ASSEMBLY_DIR" | grep "^adam-cli_[0-9A-Za-z\.-]*\.jar$" | grep -v javadoc | wc -l)"
+num_jars="$(ls -1 "$ASSEMBLY_DIR" | grep "^adam-assembly_[0-9A-Za-z\.-]*\.jar$" | grep -v javadoc | wc -l)"
 if [ "$num_jars" -eq "0" ]; then
   echo "Failed to find ADAM cli assembly in $ASSEMBLY_DIR." 1>&2
   echo "You need to build ADAM before running this program." 1>&2
   exit 1
 fi
 
-ASSEMBLY_JARS="$(ls -1 "$ASSEMBLY_DIR" | grep "^adam-cli_[0-9A-Za-z\.-]*\.jar$" | grep -v javadoc || true)"
+ASSEMBLY_JARS="$(ls -1 "$ASSEMBLY_DIR" | grep "^adam-assembly_[0-9A-Za-z\.-]*\.jar$" | grep -v javadoc || true)"
```
Comment (heuermh) [inline bin/adam-submit]: I wonder if it might be preferable to use `<finalName>adam_2.10</finalName>` for the fat jar artifact?

Comment (fnothaft) [inline bin/adam-submit]: I'm agnostic. LMK your preference.

**Comment by fnothaft** [comment]
Got the `finalName` patch in.

**Comment by heuermh** [comment]
Merged commit 48302d9772b8de18d67a0dc8ecd3014ed6133aec

Thank you, @fnothaft!

**Pull request closed without merging**